examples/go-kprobe-counter/bpf/kprobe_counter.c

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+// Copyright Authors of bpfman
+
+#include <linux/bpf.h>
+#include <linux/pkt_cls.h>
+#include <linux/types.h>
+#include <signal.h>
+
+#include <bpf/bpf_helpers.h>
+
+struct datarec {
+  __u64 counter;
+} datarec;
+
+struct {
+  __uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+  __type(key, __u32);
+  __type(value, datarec);
+  __uint(max_entries, 1);
+  __uint(pinning, LIBBPF_PIN_BY_NAME);
```

Comment:
Hrm I need to revisit the current requirement we have for specifying pin by name, I don't want our users to have to do this.

Reply:
i.e https://github.com/aya-rs/aya/issues/837

Reply:
I copied this from the tracepoint example.  What is the preferred way to pin?